source/dub/dub.d

```diff
@@ -1260,17 +1243,28 @@ class Dub {
 		settings.config = "application";
 		settings.buildType = "debug";
 		settings.compiler = getCompiler(compiler_binary);
-		settings.platform = settings.compiler.determinePlatform(settings.buildSettings, compiler_binary, m_defaultArchitecture);
-		if (m_defaultLowMemory) settings.buildSettings.options |= BuildOption.lowmem;
-		if (m_defaultEnvironments) settings.buildSettings.addEnvironments(m_defaultEnvironments);
-		if (m_defaultBuildEnvironments) settings.buildSettings.addBuildEnvironments(m_defaultBuildEnvironments);
-		if (m_defaultRunEnvironments) settings.buildSettings.addRunEnvironments(m_defaultRunEnvironments);
-		if (m_defaultPreGenerateEnvironments) settings.buildSettings.addPreGenerateEnvironments(m_defaultPreGenerateEnvironments);
-		if (m_defaultPostGenerateEnvironments) settings.buildSettings.addPostGenerateEnvironments(m_defaultPostGenerateEnvironments);
-		if (m_defaultPreBuildEnvironments) settings.buildSettings.addPreBuildEnvironments(m_defaultPreBuildEnvironments);
-		if (m_defaultPostBuildEnvironments) settings.buildSettings.addPostBuildEnvironments(m_defaultPostBuildEnvironments);
-		if (m_defaultPreRunEnvironments) settings.buildSettings.addPreRunEnvironments(m_defaultPreRunEnvironments);
-		if (m_defaultPostRunEnvironments) settings.buildSettings.addPostRunEnvironments(m_defaultPostRunEnvironments);
+		settings.platform = settings.compiler.determinePlatform(
+			settings.buildSettings, compiler_binary, this.defaultArchitecture);
+		if (this.m_config.defaultLowMemory)
```

Comment:
shouldn't we also make the `if`s use the public getters?

Reply:
Tunnel vision. Done.